### PR TITLE
Initial test-setup

### PR DESCRIPTION
**Pull request by Thor77**
* removed old test-files
* added `.travis.yml` to run the tests automatically
* removed now unused dependencies
* added coverage-checks

If you want to merge this, please activate [Travis](https://travis-ci.org/) and [coveralls](https://coveralls.io/) for this repository.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+language: python
+python:
+  - "2.7"
+  - "3.4"
```
Comment (ThomasWaldmann) [inline .travis.yml]: what about 3.3?

File: bepasty/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+import pytest
+
+from bepasty.app import create_app, create_storage
+
+
+@pytest.fixture(scope='module')
+def testclient(request):
+    '''
+    creates a Flask-testclient instance for bepasty
+    '''
```
Comment (ThomasWaldmann) [inline bepasty/tests/conftest.py]: please use triple-double quotes for docstrings, some tools have issues with triple-single.

File: bepasty/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    app = create_app()
+    db_file, app.config['DATABASE'] = mkstemp()
```
Comment (ThomasWaldmann) [inline bepasty/tests/conftest.py]: py.test offers tmpfile and some other tmp related fixtures - can you use them?

File: bepasty/tests/conftest.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    # setup a secret key
+    app.config['SECRET_KEY'] = str(random())
+    # setup permissions
+    app.config['PERMISSIONS'] = {
+        'l': 'list',
+        'c': 'create',
+        'r': 'read',
+        'd': 'delete',
+        'a': 'admin'
+    }
```
Comment (ThomasWaldmann) [inline bepasty/tests/conftest.py]: each password just giving one permission is a bit unusual. but if it works for the tests...

File: bepasty/tests/conftest.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    def teardown():
+        close(db_file)
+        unlink(app.config['DATABASE'])
+    request.addfinalizer(teardown)
```
Comment (ThomasWaldmann) [inline bepasty/tests/conftest.py]: have a look at yield_fixture, it's nice.

File: bepasty/tests/conftest.py (hunk 4 above)
Comment (Thor77) [inline bepasty/tests/conftest.py]: do you have another/better suggestion?

Comment (ThomasWaldmann) [inline bepasty/tests/conftest.py]: having sets of permissions with incrementing power?
their names could be either something verbosely readable or just their sorted first letters combined.

**Comment by ThomasWaldmann** [comment]
a general question: you killed a lot of tests without replacement, why did you do that?

**Comment by Thor77** [comment]
I just *killed* all tests, because I want to have all tests in one location (`bepasty/tests/`) and generally thought they're deprecated. But it seems like at least some of them are still up2date, should i re-add all or just some of them?

**Comment by ThomasWaldmann** [comment]
Well, maybe first try to run the tests before you kill them.
I am not aware that there were major problems with them.

File: bepasty/tests/conftest.py (hunk 4 above)
Comment (Thor77) [inline bepasty/tests/conftest.py]: I already wrote one [permission-test](https://github.com/Thor77/bepasty-server/blob/permission_tests/bepasty/tests/test_permissions.py) and just wanted to have a simple way to test the permissions one-by-one, it wouldn't be a big problem to use permissions with incrementing power, but i just didn't need it.

File: bepasty/tests/conftest.py (hunk 3 above)
Comment (Thor77) [inline bepasty/tests/conftest.py]: The `tmpdir`-fixture is [function-scoped](https://travis-ci.org/Thor77/bepasty-server/jobs/87001685#L230) (maybe there's still a way to use it, but due the lack of experience with py.test I've no idea) and the `tmpdir_factory`-fixture is [not available](https://travis-ci.org/Thor77/bepasty-server/jobs/87002362#L265) for the latest pytest-version for py2.7

**Comment by Thor77** [comment]
Seems like all the tests are still working.
My [travis-builds](https://travis-ci.org/Thor77/Thor77/bepasty-server/builds/87019261) pass now \o/

**Comment by Thor77** [comment]
Maybe you want to insert the travis and coveralls-badge into README?

**Comment by ThomasWaldmann** [comment]
Yes, would be nice (at the end, so it is less annoying for people reading it with "less").

**Pull request merged**